### ISSUE TITLE
nx show projects hangs with @jnxplus/nx-maven:0.24.3 and nx:17.0.2

### ISSUE THREAD

**Issue by mpsanchis** [khalilou88/jnxplus#810]
Hello!

I had just upgraded to `@jnxplus/nx-maven:0.24.3` and am running `npx nx show projects`. This command just hangs. Downgrading to `0.14.1` works. I am going to try versions in between to try to give more information.

I was running the command previously in a pod with few processes, and it seemed that several `java` processes were spawned by Nx. It looked like there was one per project, since at some point I saw a `-pl :<project-name>`.
I am now running it in my machine (with many more processes) and I can't spot the same.

I will continue the debugging to try to provide you with more information, but I wanted to post already to see if this behaviour rings a bell.

Best,

**Comment by jbadeau** [comment]
I see there are calls to mvn help:evaluate which will spawn multiple jvms when done in a loop. If that’s the issue maybe it can be optimized by minimizing the jvm resources via processor count arg? Another way may be to get the graph all at once and post process it.

**Comment by jbadeau** [comment]
Ahh, I see it’s a sync call so maybe it’s related to the speed of launching. Will the jvm resources be feed fast enough  immediately?

**Comment by khalilou88** [comment]
Do you think we can try to disable NX_DAEMON for this?

```
execSync(
      `${getExecutable()} help:evaluate -Dexpression=settings.localRepository -q -DforceStdout`,
      {
        cwd: mavenRootDirAbsolutePath,
        env: { ...process.env, NX_DAEMON: 'false' },
        windowsHide: true,
      },
    )
      .toString()
      .trim();
```

**Comment by mpsanchis** [comment]
Adding datapoints to this problem:
With `@jnxplus/nx-maven:0.22.10` and `nx:17.0.2`, the command `npx nx graph` takes around 2 minutes to load in a mid-sized repo (40 POM files, many of which correspond to only aggregator POMs) in a powerful MacBook M2 machine. Same machine took 1s to load the graph with `0.14.1`.
Regarding other versions, it seems that:
* `0.19.0` is still fast
* `0.20.0` spits out `Cannot read properties of undefined (reading 'outputs')`
* `0.21.0` is still fast
* `0.22.0` is still fast
* `0.23.0` is slow

**Comment by khalilou88** [comment]
Do you use static versions or with variables ?

**Comment by khalilou88** [comment]
Maybe there is no issue here, just we need to do things efficiently.
If you are using "Maven CI Friendly Versions", we should calculate version just once and not by project.

Could you please use this project : https://github.com/khalilou88/jnxplus-examples

**Comment by jbadeau** [comment]
We use ci friendly revision properties

**Comment by khalilou88** [comment]
@jbadeau @mpsanchis I published version 0.24.4 to avoid using `mvn help:evaluate` as possible.
The `revision` property should set in the root pom.xml.

If the issue persists, I need to know how you use the plugin.

**Comment by mpsanchis** [comment]
@khalilou88 

The performance has now improved to be usable, but I think the implementation can result in invalid cache. Here's a simple sketch of how our repo looks like:

```mermaid
flowchart TD
    C{root aggregator POM}
    C -->D[parent pom A \n revision 0.0.0]
    C -->E[parent pom B \n revision 0.0.0]
    C -->F[parent pom C \n revision 0.0.0]
```

In this fix, the root pom needs a revision, which is cached. This means that if 
* the children modules are released 
* the children modules are independently versioned via `-Drevision=${VERSION}`
* the release target in Nx is chacheable with `"{options.outputDirLocalRepo}"` in its outputs,

then only the `0.0.0` version of the artifacts will be cached in `.m2/repository`.

It is not a problem for us at the moment because we're not yet caching the release target, but it might affect in the future. 
The fix would be to cache parent pom's revisions. It might be that there's only one parent pom at the root, but this is a specific case.

Thank you as always for the super fast fixes 🙏

**Comment by khalilou88** [comment]
@mpsanchis I agree and the question here if the `revision plugin` fit with nx workspace?. I believe no. Maybe `nx release` will give a better alternative.

I will publish a new fix to support `revision plugin` with a generic solution without using `mvn help:evaluate`.

**Comment by khalilou88** [comment]
I published version 0.24.5 without use of `mvn help:evaluate` to get effective versions.

**Comment by khalilou88** [comment]
@jbadeau 

Can you try this please : 
```
{
  "plugins": [
    {
      "plugin": "@jnxplus/nx-maven",
      "options": {
        "mavenRootDirectory": "",
        "localRepoRelativePath": ".m2/repository"
      }
    }
  ]
}
```

without ".".

**Comment by jbadeau** [comment]
I tried both but didn’t work either

**Comment by khalilou88** [comment]
@jbadeau I don't add dot to empty projectRoot  and nx do. will do the same as nx.

**Comment by khalilou88** [comment]
@jbadeau removing `mavenRootDirectory` from options will fix the issue as a workaround but I will publish a fix after my tests.

**Comment by khalilou88** [comment]
@jbadeau fix published in version 0.24.6
